src/rd.h

```diff
@@ -38,6 +38,9 @@
 #ifndef _POSIX_C_SOURCE
 #define _POSIX_C_SOURCE 200809L  /* for timespec on solaris */
 #endif
+#ifndef _DARWIN_C_SOURCE
```

Comment:
only do this if on darwin

Reply:
Done, but do note that this is a bit incongruent. _GNU_SOURCE is defined on all platforms but only makes sense on glibc systems, and _POSIX_C_SOURCE is defined on all platforms despite only being needed (according to comment) for Solaris. This inconsistency is fine with me, but wanted to point it out!